obstore/python/obstore/_attributes.pyi

```diff
@@ -1,4 +1,10 @@
-from typing import Literal, TypeAlias
+import sys
+from typing import Literal
+
+if sys.version_info >= (3, 10):
+    from typing import TypeAlias
+else:
+    from typing_extensions import TypeAlias
```

Comment:
You could safely do directly `from typing_extensions import TypeAlias` to avoid checking the python version twice (because typing_extensions` does it as well https://github.com/python/typing_extensions/blob/7def253cd65b3a916fcb76a6eb51428b60e1bcc8/src/typing_extensions.py#L1500-L1517

Reply:
But people on latest Python versions might not have typing_extensions installed. (we should probably mark it as a runtime dependency ourselves for earlier Python versions I guess)

Reply:
> (we should probably mark it as a runtime dependency ourselves for earlier Python versions I guess)

you should 😅 

pydantic adds it for all version https://github.com/pydantic/pydantic/blob/main/pyproject.toml#L46

Reply:
FYI it didn't fail on CI because we have a lot of tests dependencies that depends on typing-extensions https://github.com/developmentseed/obstore/blob/main/uv.lock#L281

IMO it doesn't hurt that much to add typing-extensions as a dependency (except that you won't be able to say that obstore is a zero dependency module 😬)

Reply:
#375